src/main.rs

```diff
@@ -0,0 +1,200 @@
+                    Status::Resolved => 1,
+                },
+            );
+
+            //send http put to cachet
+            let client = reqwest::Client::new();
+            let response = match client.put(&format!(
+                    "{endpoint}/api/v1/components/{component}",
+                    endpoint = match std::env::var("CACHET_BASE_URL") {
+                        Ok(val) => val,
+                        Err(_) => String::from(include_str!("cachet_endpoint.txt")),
+                    },
+                    component = alert.alerts[0].annotations.component
+                ))
+                // read "Authorization: Bearer" token into "X-Cachet_Token"
+                .header(XCachetToken(match req.headers().get("Authorization") {
+                    Some(header) => match header.to_str() {
+                        Ok(header) => {
+                            let mut header_prefix = String::from(header);
+                            header_prefix.split_off(7) //cuts of the first 7 chars: "Bearer "
+                        },
+                        Err(_) => return Ok(HttpResponse::new(http::StatusCode::UNAUTHORIZED)),
+                    },
+                    None => {
+                        return Ok(HttpResponse::new(http::StatusCode::UNAUTHORIZED));
+                    },
+                })).json(&map).send() {
+                    Ok(res) => res,
+                    Err(err) => {
+                        error!("{}", err);
+                        return Ok(HttpResponse::new(http::StatusCode::INTERNAL_SERVER_ERROR))
+                    }
+                };
+            info!("{:?}", response);
+            Ok(HttpResponse::new(http::StatusCode::from_u16(response.status().as_u16()).unwrap()))
+        }).responder()
+}
+
+fn healthcheck(_: HttpRequest) -> &'static str {
+    "✅"
+}
+
+fn main() {
+    setup_logging(log::LevelFilter::Debug);
+    match server::new(|| {
+        App::new()
+            .middleware(Logger::default())
+            .route("/", http::Method::POST, hook)
+            .route("/health", http::Method::GET, healthcheck)
+    }).bind("0.0.0.0:8888")
+    {
+        Ok(server) => server.run(),
+        Err(err) => error!("Couldn't start server: {}", err),
+    }
+}
+
+fn setup_logging(level: log::LevelFilter) {
+    match fern::Dispatch::new()
+        .format(|out, message, record| {
+            out.finish(format_args!(
+                "[{}][{}] {}",
+                chrono::Local::now().format("%Y-%m-%d %H:%M:%S"),
+                record.level(),
+                message
+            ))
+        })
+        .level(level)
+        .chain(std::io::stdout())
+        .apply()
+    {
+        Err(_) => {
+            eprintln!("error setting up logging!");
+        }
+        _ => info!("logging set up properly"),
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AlertHook {
+    #[serde(rename = "version")]
```

Comment:
You can bulk rename by using the container attribute `#[serde(rename_all = "camelCase")]`, https://serde.rs/attr-rename.html

Reply:
Having it explicit feels safer, because of things like generatorURL vs generator_url. It is boilerplaty though.